textdb/textdb-dataflow/src/test/java/edu/uci/ics/textdb/dataflow/join/JoinDistanceHelper.java

```diff
@@ -0,0 +1,139 @@
+import edu.uci.ics.textdb.dataflow.common.KeywordPredicate;
+import edu.uci.ics.textdb.dataflow.keywordmatch.KeywordMatcherSourceOperator;
+import edu.uci.ics.textdb.storage.DataStore;
+import edu.uci.ics.textdb.storage.relation.RelationManager;
+
+public class JoinDistanceHelper {
+    
+    public static final String BOOK_TABLE_OUTER = "jointest_book_outer";
+    public static final String BOOK_TABLE_INNER = "jointest_book_inner";
+    
+    public static void createTestTables() throws TextDBException {
+        RelationManager relationManager = RelationManager.getRelationManager();
+        
+        // create the two book tables
+        relationManager.createTable(BOOK_TABLE_OUTER, "../index/jointest/book_outer/", 
+                JoinTestConstants.BOOK_SCHEMA, LuceneAnalyzerConstants.standardAnalyzerString());     
+        relationManager.createTable(BOOK_TABLE_INNER, "../index/jointest/book_inner/", 
+                JoinTestConstants.BOOK_SCHEMA, LuceneAnalyzerConstants.standardAnalyzerString());         
+    }
+    
+    public static void insertToOuter(ITuple... tuples) throws StorageException {
+        RelationManager relationManager = RelationManager.getRelationManager();
+        for (int i = 0; i < tuples.length; i++) {
+            relationManager.insertTuple(BOOK_TABLE_OUTER, tuples[i]);
+        }
+    }
+    
+    public static void insertToOuter(List<ITuple> tuples) throws StorageException {
+        RelationManager relationManager = RelationManager.getRelationManager();
+        for (ITuple tuple: tuples) {
+            relationManager.insertTuple(BOOK_TABLE_OUTER, tuple);
+        }
+    } 
+    
+    public static void insertToInner(ITuple... tuples) throws StorageException {
+        RelationManager relationManager = RelationManager.getRelationManager();
+        for (int i = 0; i < tuples.length; i++) {
+            relationManager.insertTuple(BOOK_TABLE_INNER, tuples[i]);
+        }
+    }
+    
+    public static void insertToInner(List<ITuple> tuples) throws StorageException {
+        RelationManager relationManager = RelationManager.getRelationManager();
+        for (ITuple tuple: tuples) {
+            relationManager.insertTuple(BOOK_TABLE_INNER, tuple);
+        }
+    } 
+    
+    public static void clearTestTables() throws TextDBException {
+        RelationManager relationManager = RelationManager.getRelationManager();
+        relationManager.deleteTuples(BOOK_TABLE_OUTER, new MatchAllDocsQuery());
+        relationManager.deleteTuples(BOOK_TABLE_INNER, new MatchAllDocsQuery());      
+    }
+    
+    public static void deleteTestTables() throws TextDBException {
+        RelationManager relationManager = RelationManager.getRelationManager();
+        relationManager.deleteTable(BOOK_TABLE_OUTER);
+        relationManager.deleteTable(BOOK_TABLE_INNER);
+    }
+    
+    public static KeywordMatcherSourceOperator getKeywordSource(String tableName, String query, 
```

Comment:
This helper function provides a `KeywordMatcherSourceOperator` since most of the test cases in Join use `KeywordMatcher`.